profiling/DerSPMF_test.jl

```diff
@@ -0,0 +1,70 @@
+using NonlinearEigenproblems, Random, SparseArrays, Revise, LinearAlgebra, BenchmarkTools
+import ..NEPTypes.AbstractSPMF
+struct DerSPMF{T<:AbstractMatrix,FDtype} <: AbstractSPMF{T}
+    spmf::SPMF_NEP{T}
+    fD::Matrix{FDtype}
+    σ::Number
+end
+
+## one constructor takes spmf as input and compute the derivatives
+function DerSPMF(spmf::SPMF_NEP,σ::Number,m::Int)
+      # Compute DD-matrix from get_fv(spmf)
+      TT=promote_type(typeof(σ),eltype(nep.A[1]))
+      p=length(nep.fi)
+      fD=zeros(TT,m,p)
+      # matrix for the computation of derivatives
+      SS=diagm(0 => σ*ones(TT,2m+2)) + diagm(-1 => (1:2m+1))
+      fD=zeros(2*m+2,p)
+      for t=1:p fD[:,t]=nep.fi[t](SS)[:,1] end
+      return DerSPMF(spmf,fD,σ);
+end
+
+import ..NEPCore.compute_Mlincomb
+function compute_Mlincomb(
+                    nep::DerSPMF{T,FDtype},
+                    λ::Number,
+                    V::AbstractVecOrMat,
+                    a::Vector=ones(size(V,2))) where {T,FDtype}
+
+    local n,k,p
+    p=size(nep.fD,2)
+    n,k=size(V)
+    # Type logic
+    TT=promote_type(eltype(V),typeof(λ),eltype(nep.spmf.A[1]),eltype(a))
+    z=zeros(TT,n)
+    afD=a.*nep.fD[1:k,:];
+    @inbounds for j=1:p
+        z .+= nep.spmf.A[j]*(V*afD[:,j])
+    end
```

Comment:
Would it be faster to also precompute `VafD=V*afD` before the for-loop?  Matrix matrix products are in general faster.